lib/auth.coffee

```diff
@@ -98,8 +98,8 @@ getAuth = (deps, opts) ->
 	###
 	exports.whoami = (callback) ->
 		getUserDetails()
-		.then (userDetails) -> userDetails.username || undefined
-		.catchReturn(undefined)
+		.then (userDetails) -> userDetails?.username || undefined
+		.catchReturn(new errors.BalenaNotLoggedIn, undefined)
```

Comment:
Is the `new` keyword misplaced here? (There's no `new` in line 234 below.)

Reply:
Nice catch @pdcastro :+1: